### PR TITLE
test: webpack cache deps test to file deps test

### PR DESCRIPTION
**Pull request by jerrykingxyz**
## Summary

Disable cache dependency case due to unsupported 'module.unsafe' and 'snapshot.managedPaths'

### <samp>🤖 Generated by Copilot at c1b8e01</samp>

Refactor and simplify file dependencies test cases for webpack. Move and rename files from `cache-dependencies` to `file-dependencies` and remove irrelevant checks and files.

<details open=true>
  <summary><h2>Walkthrough</h2></summary>

### <samp>🤖 Generated by Copilot at c1b8e01</samp>

*  Delete files and rename files related to cache-dependencies test cases, as they are replaced by file-dependencies test cases (F1,F2,F3,F4,F5,F6,F7,F8)
* Remove `snapshot` configuration from `webpack.config.js` for file-dependencies test cases, as it is not needed for caching ([link](https://github.com/web-infra-dev/rspack/pull/3445/files?diff=unified&w=0#diff-66153091b6184c6647b216eace9bf16ed9f88e48f27126e3e55c4205b5c4e847L5-L7))
* Remove assertion for `package.json` file dependency from `webpack.config.js` for file-dependencies test cases, as it is not true for module resolution ([link](https://github.com/web-infra-dev/rspack/pull/3445/files?diff=unified&w=0#diff-66153091b6184c6647b216eace9bf16ed9f88e48f27126e3e55c4205b5c4e847L18-R19))
* Remove unused variables `modules` and `children` from stats object in `ConfigTestCases.template.js`, as they are not needed for test cases ([link](https://github.com/web-infra-dev/rspack/pull/3445/files?diff=unified&w=0#diff-2d74cc353a05727a74107d4db7498a9668eff9b9ae1658078afecec67ffb23f1L250-R250))
* Comment out code that checks if all modules are cached in `ConfigTestCases.template.js`, as it is not relevant for file-dependencies test cases ([link](https://github.com/web-infra-dev/rspack/pull/3445/files?diff=unified&w=0#diff-2d74cc353a05727a74107d4db7498a9668eff9b9ae1658078afecec67ffb23f1L265-R285))

</details>

### CODE REVIEW

**Review by hyf0 — APPROVED** [review]

File: packages/rspack/tests/configCases/split-chunks/max-size-casing/webpack.config.js [hunk 1]
```diff
@@ -7,7 +7,6 @@ module.exports = {
 		newSplitChunks: true
 	},
 	optimization: {
-		chunkIds: "named",
```
Comment (ahabhgk) [inline packages/rspack/tests/configCases/split-chunks/max-size-casing/webpack.config.js:10]: Why remove this?

Comment (jerrykingxyz) [reply packages/rspack/tests/configCases/split-chunks/max-size-casing/webpack.config.js:10]: https://github.com/web-infra-dev/rspack/actions/runs/5187108757/jobs/9349855261?pr=3445

File: webpack-test/configCases/cache-dependencies/managed-items-unsafe-cache/test.filter.js [hunk 2]
```diff
@@ -1 +1,4 @@
-module.exports = () => {return false}
\ No newline at end of file
+module.exports = () => {
+	// module.unsafeCache and snapshot.managedPaths not support
+	return -1;
```
Comment (IWANABETHATGUY) [inline webpack-test/configCases/cache-dependencies/managed-items-unsafe-cache/test.filter.js:3]: -1 means will not support

File: webpack-test/ConfigTestCases.template.js [hunk 3]
```diff
@@ -262,27 +262,27 @@ const describeCases = config => {
...
-												new Error(
-													`Some modules were not cached:\n${stats.toString({
-														all: false,
-														modules: true,
-														modulesSpace: 100
-													})}`
-												)
-											);
-										}
+//										const allModules = children
```
Comment (IWANABETHATGUY) [inline webpack-test/ConfigTestCases.template.js:265]: Please explain why did you comment out this block.

**Review by IWANABETHATGUY — APPROVED** [review]

**Pull request merged**